### PR TITLE
7815.master

### PR DESCRIPTION
**Pull request by ivandudardb**
Utilized the Config Pages module to establish configurations for the copyright and social media blocks. Explored various methods to retrieve values from the configuration and display the information within the blocks, referring to relevant documentation for guidance. Integrated a link to the global configurations settings within the block configuration interface. Installed a package of Multilingual modules to facilitate language management. Experimented with the functionalities of the multilingual modules, including adding the Ukrainian language to the website and implementing the language switcher. Attempted to translate select content to assess the effectiveness of the multilingual setup.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -28,6 +29,7 @@
         "drupal/entity_pager": "^1.0",
         "drupal/facets": "^2.0",
         "drupal/fontawesome": "^2.26",
+        "drupal/geofield": "^1.57",
```
Comment (ivandudardb) [inline composer.json:32]: was missed in the previous task

File: web/modules/custom/copyright/src/Plugin/Block/Copyright.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * Provides a 'CopyrightBlock' block.
+ */
+#[Block(
+  id: "copyright_block",
+  admin_label: new TranslatableMarkup("Custom Copyright block"),
+)]
+class Copyright extends BlockBase implements ContainerFactoryPluginInterface {
+
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, protected ConfigPagesLoaderService $configPagesLoaderService, protected EntityTypeManager $entityTypeManager) {
```
Comment (KuzyaWkk) [inline web/modules/custom/copyright/src/Plugin/Block/Copyright.php:23]: Missing comment in __construct

**Review by KuzyaWkk — CHANGES_REQUESTED** [review]

**Review by KuzyaWkk — APPROVED** [review]
Looks good

**Review by lavrenchuk-dev — APPROVED** [review]

File: web/modules/custom/copyright/src/Plugin/Block/Copyright.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $storage = $this->configPagesLoaderService->load('global_configurations');
+    $copyrightField = $storage->get('field_copyright')->view('default');
+    $copyrightField['#title'] = '';
+    $copyrightField['#cache']['tags'][] = 'config_pages:1';
```
Comment (andriy-tkachuk) [inline web/modules/custom/copyright/src/Plugin/Block/Copyright.php:61]: Why?

File: web/modules/custom/copyright/src/Plugin/Block/Copyright.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $storage = $this->configPagesLoaderService->load('global_configurations');
+    $copyrightField = $storage->get('field_copyright')->view('default');
+    $copyrightField['#title'] = '';
```
Comment (andriy-tkachuk) [inline web/modules/custom/copyright/src/Plugin/Block/Copyright.php:60]: Why?

File: web/themes/custom/customtheme/customtheme.theme [hunk 5]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Form\FormStateInterface;
```
Comment (andriy-tkachuk) [inline web/themes/custom/customtheme/customtheme.theme:4]: Do you need this changes?

File: web/themes/custom/customtheme/templates/block/block--system-menu-block--social-links-footer.html.twig [hunk 6]
```diff
@@ -32,16 +32,10 @@
...
 <nav role="navigation" aria-labelledby="{{ heading_id }}"{{ attributes|without('role', 'aria-labelledby') }} class="footer-links">
-  {# Label. If not displayed, we still provide it for screen readers. #}
-  {% if not configuration.label_display %}
-    {% set title_attributes = title_attributes.addClass('visually-hidden') %}
-  {% endif %}
-  {{ title_prefix }}
-  <h2{{ title_attributes.setAttribute('id', heading_id) }}>{{ configuration.label }}</h2>
-  {{ title_suffix }}
-
+  <h2{{ title_attributes.addClass('social-title') }}>{{ config[0] }}</h2>
```
Comment (andriy-tkachuk) [inline web/themes/custom/customtheme/templates/block/block--system-menu-block--social-links-footer.html.twig:38]: are you sure config[0] exist?

Comment (andriy-tkachuk) [reply web/themes/custom/customtheme/templates/block/block--system-menu-block--social-links-footer.html.twig:38]: h2?

File: web/modules/custom/copyright/src/Plugin/Block/Copyright.php (hunk 4 above)
Comment (ivandudardb) [reply web/modules/custom/copyright/src/Plugin/Block/Copyright.php:60]: > Why?

Should I unset the title more correctly? Or get the copyright field without title?

**Pull request merged**